plugins/inputs/nstat/nstat.go

```diff
@@ -0,0 +1,183 @@
+package nstat
+
+import (
+	"bytes"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var (
+	zeroByte    = []byte("0")
+	newLineByte = []byte("\n")
+	colonByte   = []byte(":")
+)
+
+type Nstat struct {
+	ProcNetNetstat string `toml:"proc_net_netstat"`
+	ProcNetSNMP    string `toml:"proc_net_snmp"`
+	ProcNetSNMP6   string `toml:"proc_net_snmp6"`
+	DumpZeros      bool   `toml:"dump_zeros"`
+}
+
+var sampleConfig = `
+	# file paths
+	proc_net_netstat 	= 	"/proc/net/netstat"
```

Comment:
Here I think we should do the same thing nstat does: https://github.com/shemminger/iproute2/blob/master/misc/nstat.c#L46,

which is:

1. Use overrides here (but default to "")
2. try PROC_NET_NETSTAT env var
3. try PROC_ROOT env var + "net/netstat"
4. use "/proc/net/netstat"